### DISCUSSION TITLE
Question: Is there a typo in the paper for the weighting of sub-datasets?

### DISCUSSION THREAD

**Discussion by HALF111** [SalesforceAIResearch/uni2ts#55] [Q&A]
In the paper, my understanding is that to deal with data imbalance, it is guaranteed that the contribution proportion of each sub-dataset does not exceed 0.001, even though it has numerous samples; And if the samples are few, its contribution proportion corresponds to the actual sample numbers. 
If so, I'm wondering if the formula should be: $min(\frac{|D_k|}{\sum{|D_i|}}, \epsilon)$.
<img width="318" alt="b3de338ae6db571b23175b3c24ce519" src="https://github.com/SalesforceAIResearch/uni2ts/assets/58055693/d8f7362b-6743-4bfb-9922-4bb87432bcde">
Furthermore, I have questions about the partitioning of the sub-datasets - i.e., according to what criteria are the sub-datasets divided? Is it based on the domains and frequency?

**Comment by gorold** [comment]
You're absolutely right, there is a typo in the paper - see [this](https://github.com/SalesforceAIResearch/uni2ts/blob/main/example/lotsa_v1_weighting.ipynb) with the corrected equations.

A sub-dataset is simply each data source. LOTSA is a collection of many different open source time series datasets. For the notation in the paper, we call LOTSA the dataset and each component data source is called the sub-dataset.

**Comment by HALF111** [comment]
I get it. Thanks for your answer!
By the way, I found that in "cli/conf/pretrain/data/lotsa_v1_weighted.yaml", each dataset is given a weight ranging from 1e-2 to 1e2, I'm wondering how these weights are obtained and calculated. And do these weights have the same meaning as p(D_k) in the paper?

**Comment by gorold** [comment]
The weights in the yaml file have a different meaning from $p(D_k)$. You can check out the jupyter notebook linked above for more explanation on it and how it is calculated.

The short explanation is that it is required to reweight each PyTorch Dataset to achieve the sampling proportion as presented in the paper.

**Comment by HALF111** [comment]
Yes, I found the calculation of the weights in your provided notebook. Thanks for your great work!
